File: JAVA/KPVGIT/app/src/main/java/com/davidkpv/kpvgit/MainActivity.java
package com.davidkpv.kpvgit;

import androidx.appcompat.app.AppCompatActivity;


import android.os.Bundle;
import androidx.fragment.app.Fragment;
import com.davidkpv.kpvgit.ui.common.NavigationController;

import javax.inject.Inject;

import dagger.android.DispatchingAndroidInjector;
import dagger.android.support.HasSupportFragmentInjector;

public class MainActivity extends AppCompatActivity implements HasSupportFragmentInjector {

    @Inject
    DispatchingAndroidInjector<Fragment> dispatchingAndroidInjector;

    @Inject
    NavigationController navigationController;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // SI SE INICIA LA ACTIVITY POR PRIMERA VEZ INICIAMOS LA BÚSQUEDA
        if(savedInstanceState == null){
            navigationController.navigateToSearch();
        }
    }

    @Override
    public DispatchingAndroidInjector<Fragment> supportFragmentInjector(){
        return dispatchingAndroidInjector;
    }
}

File: KOTLIN/ExampleViewModel/app/src/main/java/mx/com/davidkpv/exampleviewmodel/MainActivity.kt
package mx.com.davidkpv.exampleviewmodel

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.squareup.picasso.Picasso
import mx.com.davidkpv.exampleviewmodel.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // DataBinding
        val binding = DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)

        // ViewModel
        var model = ViewModelProvider(this).get(MainActivityViewModel::class.java)
        var urlImage : MutableLiveData<String?>? = model.callUrlImage()

        urlImage?.observe(this, Observer {
            // Charge the Image
            Picasso.get().load(it).into(binding.imageView)
        })

        binding.button.setOnClickListener {
            model.randomUrl()
        }
    }
}